BUILD.bazel

```diff
@@ -0,0 +1,166 @@
+load("@npm_bazel_typescript//:index.bzl", "ts_project", "ts_config")
+load("@npm_bazel_labs//:index.bzl", "protobufjs_ts_library")
+
+proto_library(
+    name = "google_protobuf_protos",
+    srcs = glob(["node_modules/google-gax/build/protos/google/protobuf/*.proto"]) +
+    [
+        "node_modules/google-gax/build/protos/google/protobuf/compiler/plugin.proto",
+    ],
+    strip_import_prefix = "node_modules/google-gax/build/protos",
+)
+
+proto_library(
+    name = "google_rpc_protos",
+    deps = [
+        ":google_protobuf_protos",
+    ],
+    srcs = glob(["node_modules/google-gax/build/protos/google/rpc/*.proto"]),
+    strip_import_prefix = "node_modules/google-gax/build/protos/",
+)
+
+proto_library(
+    name = "google_api_protos",
+    deps = [
+        ":google_protobuf_protos",
+    ],
+    srcs = glob(["node_modules/google-gax/build/protos/google/api/*.proto"]),
+    strip_import_prefix = "node_modules/google-gax/build/protos",
+)
+
+proto_library(
+    name = "google_longrunning_protos",
+    deps = [
+        ":google_protobuf_protos",
+        ":google_api_protos",
+        ":google_rpc_protos",
+    ],
+    srcs = ["node_modules/google-gax/build/protos/google/longrunning/operations.proto"],
+    strip_import_prefix = "node_modules/google-gax/build/protos",
+)
+
+proto_library(
+    name = "service_config_proto",
+    deps = [
+        ":google_protobuf_protos",
+        ":google_rpc_protos",
+    ],
+    srcs = ["protos/service_config.proto"],
+    strip_import_prefix = "protos/",
+)
+
+protobufjs_ts_library(
+    name = "protos",
+    deps = [
+        ":google_api_protos",
+        ":google_longrunning_protos",
+        ":service_config_proto",
+    ]
+)
+
+filegroup(
+    name = "templates",
+    srcs = glob(["templates/**/*.*", "templates/**/.*"]),
+)
+
+filegroup(
+    name = "test-fixtures",
+    srcs = glob(["test-fixtures/**/*.*"]),
+)
+
+filegroup(
+    name = "baselines",
+    srcs = glob(["baselines/**/*.*", "baselines/**/.*"]),
+)
+
+npm_runtime_dependencies = [
+    "@npm//file-system",
+    "@npm//fs-extra",
+    "@npm//get-stdin",
+    "@npm//google-gax",
+    "@npm//js-yaml",
+    "@npm//module-alias",
+    "@npm//nunjucks",
+    "@npm//object-hash",
+    "@npm//protobufjs",
+    "@npm//yargs",
+]
+
+npm_test_dependencies = npm_runtime_dependencies + [
+    "@npm//assert-rejects",
+    "@npm//c8",
+    "@npm//codecov",
+    "@npm//espower-typescript",
+    "@npm//sinon",
+    "@npm//mocha",
+]
+
+npm_compile_dependencies = npm_test_dependencies + [
+    "@npm//@types/js-yaml",
+    "@npm//@types/fs-extra",
+    "@npm//@types/get-stdin",
+    "@npm//@types/mocha",
+    "@npm//@types/module-alias",
+    "@npm//@types/node",
+    "@npm//@types/nunjucks",
+    "@npm//@types/object-hash",
+    "@npm//@types/sinon",
+    "@npm//@types/yargs",
+    "@npm//gts",
+]
+
+ts_project(
+    name = "compile",
+    srcs = glob([
+        "src/**/*.ts",
```

Comment:
Is this effectively copying what's happening in `tsconfig.json`? Do we need to coordinate changes?

Reply:
TypeScript support is really wobbly in bazel :)  They use `tsconfig.json` to get some compiler options from there, but the tasks must still specifically define their inputs so that bazel could've built a dependency tree (think `make` on steroids).